examples/core/dof/app.js

```diff
@@ -0,0 +1,433 @@
+            color += texelFetch(uColor, sampleCoord, 0);
+
+            ++count;
+        }
+
+        color /= count;
+    } else {
+        color = texelFetch(uColor, fragCoord, 0);
+    }
+
+    fragColor = color;
+}
+`
+const QUAD_VERTS = [1, 1, 0,  -1, 1, 0,  1, -1, 0,  -1, -1, 0]; // eslint-disable-line
+const NUM_ROWS = 5;
+const BOXES_PER_ROW = 20;
+const NUM_BOXES = BOXES_PER_ROW * NUM_ROWS;
+const NEAR = 0.1;
+const FAR = 10.0;
+const FOCAL_LENGTH = 2.0;
+const FOCUS_DISTANCE = 3.0;
+const MAGNIFICATION = FOCAL_LENGTH / Math.abs(FOCUS_DISTANCE - FOCAL_LENGTH);
+const FSTOP = 2.8;
+const BLUR_COEFFICIENT = FOCAL_LENGTH * MAGNIFICATION / FSTOP;
+
+let texelOffset = new Float32Array(2);
+
+export const animationLoopOptions = {
+  onInitialize: ({gl}) => {
+
+    setParameters(gl, {
+      depthTest: true,
+      depthFunc: GL.LEQUAL
+    });
+
+    const PPM = Math.sqrt(gl.drawingBufferWidth * gl.drawingBufferWidth + gl.drawingBufferHeight * gl.drawingBufferHeight) / 35;   
+    const projMat = new Matrix4();
+    const viewMat = new Matrix4().lookAt({eye: [0, 0, 8]});
+
+    ////////////////////////////
+    // Create programs
+    ////////////////////////////
+
+    const sceneProgram = new Program(gl, {
+      id: "SCENE_PROGRAM",
+      vs: SCENE_VERTEX,
+      fs: SCENE_FRAGMENT
+    });
+
+    const dofProgram = new Program(gl, {
+      id: "DOF_PROGRAM",
+      vs: QUAD_VERTEX,
+      fs: DOF_FRAGMENT,
+      uniforms: {
+        uDepthRange: [0.1, 30],
+        uFocusDistance: FOCUS_DISTANCE,
+        uBlurCoefficient: BLUR_COEFFICIENT,
+        uPPM: PPM
+      },
+    });
+
+
+    //////////////////////////////////////////
+    // Initial transforms for instanced boxes
+    //////////////////////////////////////////
+
+    let boxXforms = new Array(NUM_BOXES);
+    let boxMatrices = new Float32Array(NUM_BOXES * 16);
+    let boxI = 0;
+    for (let j = 0; j < NUM_ROWS; ++j) {
+        let rowOffset = (j - Math.floor(NUM_ROWS / 2));
+        for (let i = 0; i < BOXES_PER_ROW; ++i) {
+          let scale = [0.4, 0.4, 0.4];
+          let rotate = [-Math.random() * Math.PI, 0, Math.random() * Math.PI];
+          let translate = [-i + 2 - rowOffset, 0, -i + 2 + rowOffset];
+          boxXforms[boxI] = {
+            scale: scale,
+            translate: translate,
+            rotate: rotate,
+            matrix: new Matrix4().translate(translate).rotateXYZ(rotate).scale(scale)
+          };
+            
+          boxMatrices.set(boxXforms[boxI].matrix, boxI * 16);
+        ++boxI;
+        }
+    }
+
+
+    ////////////////////////////////////////////////
+    // Set up frambuffers. 
+    ////////////////////////////////////////////////
+
+    // Need to ensure both color and depth targets can be sampled.
+    const mainFramebuffer = new Framebuffer(gl, {
+      width: gl.drawingBufferWidth,
+      height: gl.drawingBufferHeight,
+      attachments: {
```

Comment:
Nit: Looks like both color and depth attachments have standard options, `Framebuffer` object can create these when not specified, makes code cleaner.

Reply:
The default is to create a [depth Renderbuffer](https://github.com/uber/luma.gl/blob/master/modules/core/src/webgl/framebuffer.js#L654-L661) attachment, which wouldn't work for this because I need to sample depth.